compiler/AST/build.cpp

```diff
@@ -1736,6 +1719,14 @@ buildClassDefExpr(const char* name,
                   Flag        isExtern,
                   const char* docs) {
   AggregateType* ct = toAggregateType(type);
+  if (strcmp("string", name) == 0) {
```

Comment:
Yeah, we have to do this too early in the compiler to use any of the normal means, the pragma isn't applied until after this function for example. I played with a lot of different ways to try and hook the string type, this ended up as the winner.